components/lib/divider/Divider.spec.js

```diff
@@ -2,84 +2,18 @@ import '@testing-library/jest-dom';
 import { render } from '@testing-library/react';
 import { Divider } from './Divider';
 
-describe('Divider', () => {
-    test('when component has no properties it returns with default class', () => {
-        // Arrange
-        const { container } = render(<Divider />);
-
-        // Act + Assert
-        expect(container.getElementsByClassName('p-divider p-component p-divider-horizontal p-divider-solid p-divider-left').length).toBe(1);
-    });
-    test('when layout and align as property it returns with class', () => {
-        // Arrange
-        const { container } = render(<Divider layout={'horizontal'} align={'left'} />);
-
-        // Act + Assert
-        expect(container.getElementsByClassName('p-divider p-component p-divider-horizontal p-divider-left').length).toBe(1);
-    });
-    test('when layout and align as property it returns with class', () => {
-        // Arrange
-        const { container } = render(<Divider layout={'horizontal'} align={'right'} />);
-
-        // Act + Assert
-        expect(container.getElementsByClassName('p-divider p-component p-divider-horizontal p-divider-right').length).toBe(1);
-    });
-    test('when layout and align as property it returns with class', () => {
-        // Arrange
-        const { container } = render(<Divider layout={'horizontal'} align={'center'} />);
-
-        // Act + Assert
-        expect(container.getElementsByClassName('p-divider p-component p-divider-horizontal p-divider-center').length).toBe(1);
-    });
-
-    test('when tpe has not any property it returns with default class', () => {
-        // Arrange
-        const { container } = render(<Divider />);
```

Comment:
The original test is more complicated, than it need to be. Did you notice you have the same test multiple times?

Reply:
I noticed this. But here I see the benefit of the function you wrote more than the snapshot benefit. You have created html code for all cases. The only thing we want to check is the existence of a class in render. If we created the same function for the old structure, which one would you think is advantageous? Please let me know if there's anything I missed. I will be happy to learn.

I am talking about this function:
`const snapshot = (props, name) => expect(render(<Divider {...props} />).container).toMatchSnapshot(name);`

Reply:
So the real advantage of Snapshot testing over this current test.    lets take for example this:
```js
expect(container.getElementsByClassName('p-divider p-component p-divider-horizontal p-divider-center').length).toBe(1);
```

This test will NOT fail if the following is changed:
1. If Divider is changed from a `DIV` to a `SPAN` this test would still pass and it should not the structure of the component has changed.
2. If a new style class `p-new-class` was introduced in the component this test would still pass because the current query will still be true. And to me this test should fail if a NEW class is added to the component or an unexpected one is there like `p-filled` when it should not be etc.

So the benefits to me are tremendous.